### PR TITLE
Pokemon

### PR DESCRIPTION
**Pull request by LouieBloo**

### CODE REVIEW

File: src/app/classes/game/PokemonStandard.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { GameType} from "../../interfaces/game";
+import { Game } from "./game";
+
+export class PokemonStandard extends Game {
+    override startingLifeTotal = 0;
+    override gameType:GameType = GameType.PokemonStandard;
+    override searchTag:string = "standard";
+    override coinImagePathPrefix:string = "mew"
+
+    prizeCardsToWin:number = 6;
```
Comment (domleggio) [inline src/app/classes/game/PokemonStandard.ts]: should this be readonly?

File: src/app/components/coin-flip/coin-flipper/coin-flipper.component.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+<div *ngIf="visible" class="">
```
Comment (domleggio) [inline src/app/components/coin-flip/coin-flipper/coin-flipper.component.html:2]: empty class

File: src/app/components/coin-flip/coin-flipper/coin-flipper.component.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+export class CoinFlipperComponent {
+
+  @Input() playerId!: string;
+
+  visible = false;
+  activeCoins: any = [];
+
+  private apiTimeout: any;
+  private clearCoinTimeout: any;
+  private subscriptions: Subscription = new Subscription();
```
Comment (domleggio) [inline src/app/components/coin-flip/coin-flipper/coin-flipper.component.ts:24]: is this supposed to be an array?

File: src/app/components/users/user-stream/user-stream.component.html [hunk 4]
```diff
@@ -1,19 +1,18 @@
...
-        <!-- <img class="w-full" style="max-width: fit-content;" src="https://as1.ftcdn.net/v2/jpg/01/63/11/70/1000_F_163117064_syJkTuCddASYjvl4WqyRmnuy8cDXpoQY.jpg"> -->
     </div>
 
     <div class="absolute top-1 w-full text-white  flex flex-col items-center">
-        <!-- <div>
-            <h2>{{player?.name | titlecase }}</h2>
-        </div> -->
-        <app-life-total [player]="player" [modifyCallback]="modifyLifeTotal" [editable]="localStream"
-            [modifyPoisonCallback]="modifyPoisonTotal" [modifyEnergyCallback]="modifyEnergyTotal"></app-life-total>
+        <div *ngIf="gameService.isMagicGame()">
```
Comment (domleggio) [inline src/app/components/users/user-stream/user-stream.component.html:9]: i think this gets called constantly at each change detection

File: src/app/components/users/user-stream/user-stream.component.html [hunk 5]
```diff
@@ -1,19 +1,18 @@
...
+            <app-life-total [player]="player" [modifyCallback]="modifyLifeTotal" [editable]="localStream"
+                [modifyPoisonCallback]="modifyPoisonTotal" [modifyEnergyCallback]="modifyEnergyTotal"></app-life-total>
+        </div>
+        <div *ngIf="gameService.isPokemonGame()">
```
Comment (domleggio) [inline src/app/components/users/user-stream/user-stream.component.html:13]: same here

File: src/app/components/users/user-stream/user-stream.component.html [hunk 6]
```diff
@@ -40,14 +39,27 @@ <h2>{{player?.name | titlecase }}</h2>
 
     <!-- Status bar -->
     <div class="flex flex-col absolute" id="statusBar">
-        <div *ngIf="player.hasCitiesBlessing">
-            <p class="text-center text-4xl relative"  appTooltip="This player has the Cities Blessing" [tooltipDirection]="'bottom'" [tooltipStartOpen]="false">⛪</p>
+        <!-- Magic -->
+        <div *ngIf="gameService.isMagicGame()">
```
Comment (domleggio) [inline src/app/components/users/user-stream/user-stream.component.html:43]: same here. This is also the same ngIf as line 9.

File: src/app/services/search/card-search.service.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class CardSearchService {
+  
+  private scryfallUrl = 'https://api.scryfall.com/cards/search';
+
+  constructor(private http: HttpClient) { }
+
+  searchCards(query: string, fuzzy: boolean = true, game:Game): Observable<any> {
```
Comment (domleggio) [inline src/app/services/search/card-search.service.ts:18]: can this <any> be typed?

File: src/app/services/search/card-search.service.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+
+  searchScryfall(query: string, fuzzy: boolean = true, format: string = 'commander'): Observable<any> {
+    let searchQuery = query;
+    if (fuzzy) {
+      searchQuery = `${query}`;
+    }
+    const params = new HttpParams().set('q', `${searchQuery} format=${format}`);
+    return this.http.get<any>(this.scryfallUrl, { params });
```
Comment (domleggio) [inline src/app/services/search/card-search.service.ts:33]: what is server returning here? Can we remove <any> by creating a model?

File: src/app/services/search/card-search.service.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+
+  searchPokemon(name: string, format: string = 'standard'): Observable<any> {
+    let searchQuery = `name:"*${name}*" legalities.${format}:Legal`;
+    const params = new HttpParams().set('query', searchQuery);
+    return this.http.get<any>(environment.socketUrl + "/pokemon-cards", { params });
```
Comment (domleggio) [inline src/app/services/search/card-search.service.ts:40]: same here with the any

File: src/app/classes/game/PokemonStandard.ts (hunk 1 above)
Comment (LouieBloo) [reply src/app/classes/game/PokemonStandard.ts]: fixed!

**Pull request merged**